addons/mail/static/src/models/user/user.js

```diff
@@ -35,6 +95,88 @@ function factory(dependencies) {
             return partner.nameOrDisplayName;
         }
 
+        /**
+         * Gets the canonical chat between this user and the current user.
+         *
+         * If a chat is not appropriate a notification is displayed instead.
+         *
+         * @returns {mail.thread|undefined}
+         */
+        async getCanonicalChat() {
+            if (!this.partner) {
+                await this.async(() => this.fetchPartner());
+            }
+            if (!this.partner) {
+                // Can only happen if an invalid user id is given at insert.
+                this.env.services['notification'].notify({
+                    message: this.env._t("You can only chat with existing users."),
+                    type: 'warning',
+                });
+                return;
```

Comment:
How does it happen? Isn't it better to just crash when this happens?

Reply:
Whenever a user is deleted from the server, but an open tab still had a referenced to it from before.

Reply:
I don't think it is ever better to crash, especially this one it is really simple enough to handle.

Reply:
Would clarify in comment how it's possible to keep invalid user id, because at a glance it doesn't look right to support this case.

Reply:
Not really sure how to express it but I added a comment to both.